### PR TITLE
tarball.py: update state names and put tarball URL in tarball node

### PR DESCRIPTION
**Pull request by gctucker**
Use the new Node state names and only set the tarball artifact URL in the tarball node, not the checkout node.

Depends on #112

### CODE REVIEW

File: src/tarball.py [hunk 1]
```diff
@@ -98,25 +98,24 @@ def _push_tarball(self, config, describe):
...
             'name': 'tarball',
-            'artifacts': checkout_node['artifacts'],
+            'path': checkout_node['path'] + ['tarball'],
+            'artifacts': {
+                'tarball': urllib.parse.urljoin(self._storage_url, tarball),
+            },
             'revision': checkout_node['revision'],
-            'status': status,
-            'result': result,
+            'status': 'available',
```
Comment (JenySadadia) [inline src/tarball.py:110]: I think the tarball node should be created with `running` state first. After the execution of `_push_tarball` successfully, the state of the node should be changed to `available`.

```suggestion
            'status': 'running',
```

File: src/tarball.py [hunk 2]
```diff
@@ -98,25 +98,24 @@ def _push_tarball(self, config, describe):
...
 
-    def _update_checkout_node(self, node, describe, version, tarball):
+    def _update_checkout_node(self, node, describe, version):
         node['revision'].update({
             'describe': describe,
             'version': version,
         })
-        node['artifacts'] = {
-            'tarball': urllib.parse.urljoin(self._storage_url, tarball),
-        }
```
Comment (JenySadadia) [inline src/tarball.py:119]: Why is this removed?

Comment (gctucker) [reply src/tarball.py:119]: The tarball artifact belongs in the tarball node, not the checkout node.

File: src/tarball.py (hunk 1 above)
Comment (gctucker) [reply src/tarball.py:110]: Yes, that's right.  Will update the code accordingly.

Eventually this probably should become a template to run in a local shell anyway, based on `python.jinja2`.

Comment (gctucker) [reply src/tarball.py]: OK I've added an extra commit as it's more than just changing the state names.

File: src/tarball.py [hunk 3]
```diff
@@ -98,25 +98,30 @@ def _push_tarball(self, config, describe):
...
+    def _create_tarball_node(self, checkout_node):
+        tarball_node = {
             'parent': checkout_node['_id'],
             'name': 'tarball',
-            'artifacts': checkout_node['artifacts'],
+            'path': checkout_node['path'] + ['tarball'],
             'revision': checkout_node['revision'],
-            'status': status,
-            'result': result,
+            'state': 'running',
```
Comment (JenySadadia) [inline src/tarball.py:107]: This line can be omitted as the default node state is `running`.

File: src/tarball.py [hunk 4]
```diff
@@ -98,27 +98,30 @@ def _push_tarball(self, config, describe):
...
+        return self._db.submit({'node': tarball_node})[0]
+
+    def _make_tarball_node_available(self, tarball_node, tarball):
+        tarball_node.update({
+            'state': 'available',
+            'artifacts': {
+                'tarball': urllib.parse.urljoin(self._storage_url, tarball),
+            },
+        })
+        return self._db.submit({'node': tarball_node})
```
Comment (JenySadadia) [inline src/tarball.py:117]: The return value is not being used.
```suggestion
        self._db.submit({'node': tarball_node})
```
